ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HivePushdownSnapshotFilterRule.java

```diff
@@ -113,7 +112,7 @@ private boolean setSnapShotId(RexNode op1, RexNode op2) {
       }
 
       RexLiteral literal = (RexLiteral) op1;
-      if (literal.getType().getSqlTypeName().getFamily() != SqlTypeFamily.NUMERIC) {
+      if (literal.getTypeName() != SqlTypeName.DECIMAL) {
```

Comment:
How about 
```
if (literal.getTypeName() != HiveAugmentSnapshotMaterializationRule.snapshotIdType(rexBuilder.getTypeFactory()).getSqlTypeName())
```

The method `snapshotIdType` is used when augmenting the plan with filter predicates with snapshotId.

Reply:
At the moment keeping DECIMAL seems sufficient and more or less inline with the existing behavior. If we want to make the checks more precise I guess we can discuss about such improvements under a separate JIRA/PR since its not strictly related to the upgrade.